src/util/db.rs

```diff
@@ -129,10 +166,10 @@ where
     T: FromRow<T>
```

Comment:
The `FromRow` and `RowOrder` traits are a dangerous way to implement our database query interactions -- there are a number of database queries with a large number of columns, the order of which determines how they get parsed into objects, and many fields have the same type -- meaning that simple bugs in the ordering and corresponding indexing would result in Errors in the best case, and silent bugs in the worst case. Using the column names would remove the necessity of the index argument (and types that only need _one_ column of the row like hashes, etc., should implement FromSQL rather than FromRow) and avoid the possibility of those kinds of bugs.

Reply:
:100:.  I didn't know you could index a row's fields by column in rusqlite, but it turns out [you can](https://docs.rs/rusqlite/0.20.0/rusqlite/struct.Statement.html).  I will make this change.

Reply:
Yep -- the Row::get function accepts RowIndex's which is implemented for usize and &str

Reply:
I opened issue #1188 which will track progress on this.  I don't think it would break anything to leave the database traits as-is in this PR, but I'll gladly send a separate PR to fix this.